OrbitGl/App.cpp

```diff
@@ -597,6 +597,13 @@ void OrbitApp::StopCapture() {
 void OrbitApp::ClearCapture() {
   Capture::ClearCaptureData();
 
+  // Trigger deallocation of previous sampling related data.
+  sampling_report_ = nullptr;
+  auto empty_sampling_profiler = std::make_shared<SamplingProfiler>();
+  Capture::GSamplingProfiler = empty_sampling_profiler;
```

Comment:
There might be the same issue as above due to the execution order, when ClearCapture is invoked. However, I think you spotted the right class for the Memory leak. I just looked up when `SamplingProfiler` and found this in `CaptureSerializer:
```
if (Capture::GSamplingProfiler == nullptr) {
    Capture::GSamplingProfiler = std::make_shared<SamplingProfiler>();
  }
  Capture::GSamplingProfiler->ClearCallstacks();
```
I think the if should be removed and the clear callstacks method should be also removed. And we are fine.
Looping in @IrinaShkviro, as she is currently working on moving the callstacks from sampling profiler. So might be interssting.

Reply:
I removed that block altogether since it's done in ClearCapture now.